src/hyperspectral/hyperspectral/band_selection/models.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2021 Azavea
+#
+# Redistribution and use  in source and binary forms,  with or without
+# modification, are  permitted provided that the  following conditions
+# are met:
+#
+# 1. Redistributions  of source code  must retain the  above copyright
+# notice, this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions  and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+#
+# 3. Neither  the name of  the copyright holder  nor the names  of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY  THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS"  AND ANY EXPRESS  OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT
+# LIMITED TO,  THE IMPLIED  WARRANTIES OF MERCHANTABILITY  AND FITNESS
+# FOR  A PARTICULAR  PURPOSE ARE  DISCLAIMED.  IN NO  EVENT SHALL  THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT  LIMITED TO,  PROCUREMENT OF  SUBSTITUTE GOODS  OR SERVICES;
+# LOSS OF  USE, DATA,  OR PROFITS;  OR BUSINESS  INTERRUPTION) HOWEVER
+# CAUSED AND ON  ANY THEORY OF LIABILITY, WHETHER  IN CONTRACT, STRICT
+# LIABILITY, OR  TORT (INCLUDING  NEGLIGENCE OR OTHERWISE)  ARISING IN
+# ANY WAY  OUT OF  THE USE OF  THIS SOFTWARE, EVEN  IF ADVISED  OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import numpy as np
+import torch as torch
+import torch.nn.functional as F
+
+
+def whitening_matrix(m: np.array):
```

Comment:
Note that there is an existing (but not as robust) definition of this idea [here](https://github.com/azavea/nasa-hyperspectral/blob/master/src/hyperspectral/hyperspectral/math/rpca.py#L136-L137).  Please consider moving this implementation to replace that one.

Reply:
Okay, that's a good idea.  I will move the code over there before merging :+1: